models/discordactions.js

```diff
@@ -659,6 +660,144 @@ const updateIdle7dUsersOnDiscord = async () => {
   };
 };
 
+const updateUsersWith31DaysPlusOnboarding = async () => {
+  try {
+    const allOnboardingUsers31DaysCompleted = await getUsersBasedOnFilter({
+      state: userState.ONBOARDING,
+      time: "31d",
+    });
+    const discordMembers = await getDiscordMembers();
+    const groupOnboardingRole = await getGroupRole("group-onboarding-31d+");
+    const groupOnboardingRoleId = groupOnboardingRole.role.roleid;
+    if (!groupOnboardingRole.roleExists) throw new Error("Role does not exist");
+
+    const usersAlreadyHavingOnboaring31DaysRole = [];
+
+    discordMembers?.forEach((discordUser) => {
+      const isDeveloper = discordUser.roles.includes(discordDeveloperRoleId);
+      const haveOnboarding31DaysRole = discordUser.roles.includes(groupOnboardingRoleId);
+      if (isDeveloper && haveOnboarding31DaysRole) {
+        usersAlreadyHavingOnboaring31DaysRole.push({ discordId: discordUser.user.id });
+      }
+    });
+
+    const usersForRoleAddition = allOnboardingUsers31DaysCompleted.filter(
+      (user1) => !usersAlreadyHavingOnboaring31DaysRole.some((user2) => user1.discordId === user2.discordId)
+    );
+
+    const usersForRoleRemoval = await Promise.all(
```

Comment:
here this code  block is supposed to return a list of users who have the group-onboarding-31d+ role but are no longer in the onboarding state. However, if any of the calls to fetchUser() or getUserStatus() fail, the corresponding user will be omitted from the list.

Reply:
Yes, that's right.  If any of the api fails, the corresponding user not be the part of list.
To handle it, I have now added one variable "const errorInFetchingUserDetailsForRoleRemoval = { count: 0, errors: [] };"
which will keep the track of such users along with their discordId.

Let me know if this works, or we can handle it differently.